src/editor/Image.js

```diff
@@ -0,0 +1,80 @@
+define(function (require, exports, module) {
+    "use strict";
+
+    var Caman           = require("caman");
+    var FilerFileSystem = require("fileSystemImpl");
+    var FilerUtils       = require("filesystem/impls/filer/FilerUtils");
+    var FileSystemCache = require("filesystem/impls/filer/FileSystemCache");
+    var Path            = require("filesystem/impls/filer/BracketsFiler").Path;
+    var mimeFromExt     = require("filesystem/impls/filer/lib/content").mimeFromExt;
+
+    function initializeFilterButtons(image, imagePath) {
+        var imageMimeType = mimeFromExt(Path.extname(imagePath));
+        var imageDataRegex = /base64,(.+)/;
+        var $saveBtn = $(".btn-image-filter-save");
+        var $resetBtn = $(".btn-image-filter-reset");
+
+        $resetBtn.click(function() {
+            image.reset();
+            $saveBtn.prop("disabled", true);
+            $resetBtn.prop("disabled", true);
+        });
+
+        $saveBtn.click(function() {
+            var imageBase64Data = image.canvas.toDataURL(imageMimeType);
+            var data = FilerUtils.base64ToBuffer(imageDataRegex.exec(imageBase64Data)[1]);
+
+            FilerFileSystem.writeFile(imagePath, data, {encoding: null}, function(err) {
+                if(err) {
+                    console.error("[Bramble] Image with filters failed to save with: ", err);
+                    return;
+                }
+
+                FileSystemCache.refresh(function(err) {
+                    if(err) {
+                        console.error("[Bramble] Failed to refresh filesystem cache when applying image filters with: ", err);
+                    }
+
+                    $saveBtn.prop("disabled", true);
+                    $resetBtn.prop("disabled", true);
+                });
+            });
+        });
+
+        function applyFilterFn(fnName, args) {
+            image.reset();
+            image[fnName].apply(image, args);
+            image.render();
+            $saveBtn.prop("disabled", false);
```

Comment:
@humphd I've taken most of the solution (no spinner) from @flukeout 's [pull request](https://github.com/timmoy/brackets/pull/3/commits/71871c85abc1092471aecbfdd31efa684be14084). 

I've tested it and it works well on my machine. I think this can be (or at least pretty close to being) landed?